src/query-builder/delete-query-builder.ts

```diff
@@ -172,6 +176,98 @@ export class DeleteQueryBuilder<DB, TB extends keyof DB, O>
     })
   }
 
+  /**
+   * Adds a `using` clause to the query.
+   *
+   * This clause allows adding additional tables to the query for filtering/returning
+   * only. Usually a non-standard syntactic-sugar alternative to a `where` with a sub-query.
+   *
+   * ### Examples:
+   *
+   * ```ts
+   * await db
+   *   .deleteFrom('pet')
+   *   .using('person')
+   *   .whereRef('pet.owner_id', '=', 'person.id')
+   *   .where('person.first_name', '=', 'Bob')
+   *   .executeTakeFirstOrThrow()
+   * ```
+   *
+   * The generated SQL (PostgreSQL):
+   *
+   * ```sql
+   * delete from "pet"
+   * using "person"
+   * where "pet"."owner_id" = "person"."id"
+   *   and "person"."first_name" = $1
+   * ```
+   *
+   * On supported databases such as MySQL, this clause allows using joins, but requires
+   * at least one of the tables after the `from` clause to be also named after
+   * the `using` clause. See also {@link innerJoin}, {@link leftJoin}, {@link rightJoin}
+   * and {@link fullJoin}.
+   *
+   * ```ts
+   * await db
+   *   .deleteFrom('pet')
+   *   .using('pet')
+   *   .leftJoin('person', 'person.id', 'pet.owner_id')
+   *   .where('person.first_name', '=', 'Bob')
+   *   .executeTakeFirstOrThrow()
+   * ```
+   *
+   * The generated SQL (MySQL):
+   *
+   * ```sql
+   * delete from `pet`
+   * using `pet`
+   * left join `person` on `person`.`id` = `pet`.`owner_id`
+   * where `person`.`first_name` = ?
+   * ```
+   *
+   * You can also chain multiple invocations of this method, or pass an array to
+   * a single invocation to name multiple tables.
+   *
+   * ```ts
+   * await db
+   *   .deleteFrom('toy')
+   *   .using(['pet', 'person'])
+   *   .whereRef('toy.pet_id', '=', 'pet.id')
+   *   .whereRef('pet.owner_id', '=', 'person.id')
+   *   .where('person.first_name', '=', 'Bob')
+   *   .returning('pet.name')
+   *   .executeTakeFirstOrThrow()
+   * ```
+   *
+   * The generated SQL (PostgreSQL):
+   *
+   * ```sql
+   * delete from "toy"
+   * using "pet", "person"
+   * where "toy"."pet_id" = "pet"."id"
+   *   and "pet"."owner_id" = "person"."id"
+   *   and "person"."first_name" = $1
+   * returning "pet"."name"
+   * ```
+   */
+  using<TE extends [TableExpression<DB, TB>, ...TableExpression<DB, TB>[]]>(
```

Comment:
Isn't this exactly like `selectFrom` type-wise? I think you could just copy-paste `selectFrom` types from `query-creator` instead of creating a new set of types. You don't need to rename the types either IMO.

Reply:
I remember it wasn't for some reason, will try again.